src/core/components/Paginate/index.js

```diff
@@ -7,6 +7,8 @@ import translate from 'core/i18n/translate';
 
 import './Paginate.scss';
 
+// This is the maximal number of add-ons the API can return for a query.
+const MAX_ADDONS = 25000;
```

Comment:
This limit looks like it's only applied to views that utilize Elastic Search (i.e. not views that utilize the database directly): https://github.com/mozilla/addons-server/search?utf8=%E2%9C%93&q=ES_MAX_RESULT_WINDOW&type=

Thus, we shouldn't simulate the limit for all paginated components. In fact, the paginate component may not be working with API data at all so I don't think this is the right place.